src/lib/yang/schema.lua

```diff
@@ -564,6 +570,26 @@ local primitive_types = set(
    'binary', 'bits', 'boolean', 'decimal64', 'empty', 'enumeration',
    'identityref', 'instance-identifier', 'leafref', 'string', 'union')
 
+-- Inherits config attributes from parents
+local function inherit_config(schema, env)
```

Comment:
Looks good as-is!  May I suggest one simplification: our "env" is really just one bit, "is-config".  It starts off as true, and a non-nil "schema.config" can make it go false.  So consider replacing `env` with an `is_config` argment.

Also can you add an assertion that prevents "config true" from occuring within `is_config == false`, as per RFC 6020?